src/services/XmppSocket.js

```diff
@@ -283,6 +283,20 @@ export default {
       this.context.$store.setContactPresence({ jid: fullJid.bare, presence: presence.show, status: presence.status })
     })
 
+    // listen for retracted messages
+    this.client.on('messageRetracted', (retracted) => {
+      this.context.$store.updateMessage({
+        stanzaId: retracted.stanzaId,
```

Comment:
I consider `stanzaId` is always set

Reply:
The fasten-ed id will always be a stanza Id here indeed. I am not sure what this `stanzaId` is used for though? Is it only to ensure it's set?

Reply:
It is `id` attribute that I use to find the affected message.
The question was in fact: does all messages that can be retracted have a `StanzaId`?

Reply:
Ok I just realized you were calling `updateMessage` with these attributes! :)
I can only guess what some correspond to but not all. I guess it is fine?

If the MUC supports this, then it should also set a `stanzaId` I assume, yes.